src/main/java/org/gwtproject/tools/cldr/LocaleData.java

```diff
@@ -15,1117 +15,1276 @@
 
-  private Map<MapKey, Map<String, String>> maps;
+        return map;
+    }
+
+
+    public Map<String, String> getRegionCurrencyEntries(InputFactory cldrFactory) {
+        InputFile supp = cldrFactory.getSupplementalData();
+        XPathParts parts = new XPathParts();
+        Map<String, String> regionCurrencyMap = new HashMap<>();
+        for (String path : supp.listPaths("//supplementalData/currencyData/region")) {
+            parts.set(supp.getFullXPath(path));
+            Map<String, String> attr = parts.findAttributes("currency");
+            if (attr == null) {
+                continue;
+            }
+            String region = parts.findAttributeValue("region", "iso3166");
+            String curCode = attr.get("iso4217");
+            if ("ZZ".equals(region) || "false".equals(attr.get("tender")) || "XXX".equals(curCode)) {
+                // ZZ is an undefined region, XXX is an unknown currency code (and needs
+                // to be special-cased because it is listed as used in Anartica!)
+                continue;
+            }
+            String to = attr.get("to");
+            if (to == null) {
+                regionCurrencyMap.put(region, curCode);
+            }
+        }
+        return regionCurrencyMap;
+    }
+
+    public Map<String, String> getLanguageMostPopulatedRegion(InputFactory cldrFactory) {
+        Map<String, String> localesRegions = new HashMap<>();
+        InputFile supp = cldrFactory.getSupplementalData();
+        XPathParts parts = new XPathParts();
+        Map<String, Double> languageMaxPopulation = new HashMap<>();
+        for (String path : supp.listPaths("//supplementalData/territoryInfo/territory")) {
+
+            try {
+                parts.set(supp.getFullXPath(path));
+
+                Map<String, String> territoryAttributes = parts.findAttributes("territory");
+                if (territoryAttributes == null) {
+                    continue;
+                }
+
+                String region = territoryAttributes.get("type");
+                if (!"ZZ".equals(region)) {
+                    double population = Double.parseDouble(territoryAttributes.get("population"));
+                    Map<String, String> languagePopulationAttributes = parts.findAttributes("languagePopulation");
+                    String language = languagePopulationAttributes.get("type");
+                    double populationPercent = Double.parseDouble(languagePopulationAttributes.get("populationPercent"));
+
+                    Double languagePopulation = (populationPercent / 100) * population;
+
+                    if (!languageMaxPopulation.containsKey(language) || languagePopulation > languageMaxPopulation.get(language)) {
+                        languageMaxPopulation.put(language, languagePopulation);
+                        localesRegions.put(language, region);
+                    }
+                }
+            } catch (Exception ex) {
+                System.out.println(ex);
```

Comment:
again, should we just continue here? if this is important, should use error/warn loggers (and possibly capture stack trace)?